perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2459,6 +2462,17 @@ public void updateFormItem(PerunSession sess, ApplicationFormItem item) throws P
 			}
 		}
 
+		// pending vo data for current user (if known to Perun) or session principal
+		if (group != null) {
+			Optional<Application> pendingVoApplication = getApplicationsForUser(sess)
+				.stream()
+				.filter(voApp -> voApp.getGroup() == null)
+				.findFirst();
```

Comment:
I think you are missing a filter for states NEW and VERIFIED. Now it could return REJECTED/DELETED app

Reply:
For sure, thanks. Added.